exe/wallet/http-bridge/Main.hs

```diff
@@ -106,144 +122,186 @@ import qualified Network.Wai.Handler.Warp as Warp
                               Main entry point
 -------------------------------------------------------------------------------}
 
-main :: IO ()
-main = runCli withBackend cliDefinition
-
-withBackend :: Manager -> Environment -> IO ()
-withBackend manager env@Environment {..} = do
-    listen <- parseWalletListen env
-    backendPort <- parseArg $ longOption "backend-port"
-    dbFile <- parseOptionalArg $ longOption "database"
-    stateDir <- parseOptionalArg $ longOption "state-dir"
-    let verbosity = verbosityFromArgs args
-    -- FIXME
-    -- We also need to support the special arg 'local' for the bridge... or not.
-    parseArg (longOption "network") >>= \case
-        Testnet ->
-            runCliCommand env manager
-                (execServe @'Testnet verbosity listen backendPort dbFile)
-                (execLaunch "testnet" verbosity listen backendPort stateDir)
-        Mainnet ->
-            runCliCommand env manager
-                (execServe @'Mainnet verbosity listen backendPort dbFile)
-                (execLaunch "mainnet" verbosity listen backendPort stateDir)
+main :: forall (n :: Network). IO ()
+main = join $ execParser $ cli $ mempty
+    <> cmdLaunch
+    <> cmdServe
+    <> cmdMnemonic
+    <> cmdWallet @(HttpBridge n)
+    <> cmdTransaction @(HttpBridge n)
+    <> cmdAddress @(HttpBridge n)
+    <> cmdVersion
 
```

Comment:
Calling `cardano-wallet <noun> --help` gives help that is relevant, but `cardano-wallet --help <noun>` (notice that the order is switched) doesn't produce what you'd expect. For example:
```
$ cardano-wallet -- --help mnemonic
The CLI is a proxy to the wallet server, which is required for most commands.
Commands are turned into corresponding API calls, and submitted to an
up-and-running server. Some commands do not require an active server and can be
run offline (e.g. 'mnemonic generate').

Usage: cardano-wallet COMMAND
  Cardano Wallet Command-Line Interface (CLI)

Available options:
  -h,--help                Show this help text

Available commands:
  launch                   Launch and monitor a wallet server and its chain
                           producer.
  serve                    serve API that listens for commands/actions.
  mnemonic                 
  wallet                   
  transaction              
  address                  
  version                  Show the program's version.
```

Reply:
Hmmm.. But _what_ would you actually expects from `$ cardano-wallet -- --help mnemonic` :sweat_smile: ?

To me, it's a parsing error, and I'd expect the CLI to show me the usage of the root command because there's indeed something I didn't get about it :smile: ...

Reply:
I would expect that the following produce the same output:

```cardano-wallet --help mnemonic```
```cardano-wallet mnemonic --help```

For a similar example, see:
```git fetch --help```
```git --help fetch```

But it is a minor issue, I agree.